src/software/ai/hl/stp/tactic/shadow_enemy/shadow_enemy_fsm.cpp

```diff
@@ -111,16 +136,68 @@ void ShadowEnemyFSM::blockShot(const Update &event,
     processEvent(MoveFSM::Update(control_params, event.common));
 }
 
-void ShadowEnemyFSM::stealAndChip(const Update &event)
+void ShadowEnemyFSM::goAndSteal(const Update &event)
+{
+    auto ball_position = event.common.world_ptr->ball().position();
+    auto face_ball_orientation =
+        (ball_position - event.common.robot.position()).orientation();
+
+    std::optional<EnemyThreat> enemy_threat_opt = event.control_params.enemy_threat;
+    bool go_for_ball                            = true;
+    auto goal_direction =
+        event.common.world_ptr->field().friendlyGoalCenter() - ball_position;
+    auto enemy_angle = Vector();
```

Comment:
don't use `auto`, using the explicit times improves code readability. we generally reserve `auto` for when the type name is really long and unwieldy